### ISSUE TITLE
Add circuit breaking

### ISSUE THREAD

**Issue by deuch** [fabiolb/fabio#281]
Is it possible to add a tag in services to define some fallback node for circuit breaking ? Those node may not serve the initial requests but will respond something to ensure a degraded services instead of errors.

For example : 

service : myservice
routes : 
node-1
node-2
node-3
node-4 tag "fallback"
node-5 tag "fallback"

In nominal case, fallback node are ignored. If node1, 2 and 3 are down, requests are send to node-4 and node-5. If a nominal node (ie node-1) come back to life, requests are terminated on node-4 and node-5 and new requests are send to node-1 for example (ok the node-1 will handle all the traffic at the beginning but you see my point).

What do you think about it ?

**Comment by magiconair** [comment]
I need to check the code but what happens when you define node-4 with weight 0? If I didn't special case it then it should receive 100% of the traffic if the other services are not there. However, weight is not the right concept here since weight 0 would be counter intuitive. Sounds more like a priority concept.

**Comment by deuch** [comment]
Even if the node-4 has a 0 weight it can not be served if node-1 2 or 3 are up. Node 4 or 5 (regardless the weight) can not serve request if one of the node 1,2 or 3 is up. The main idea if it's the service can not answer, use a fallback node to have a response anyway and not a 503 or 404 error. It's like the concept of circuit breaker like hystrix. 

Normally if node 1, 2 and 3 are down, they do not appear in the routing table, so the only node that can respond are node 4 and 5. But when a normal node come back, the tag fallback must discard node 4 and 5 to serve request and the priority is done to normal nodes.

Does it make sense to you ?

**Comment by magiconair** [comment]
Yes, I understand what you're looking for. I was thinking whether `weight 0` has the effect you're looking for as an unintended side-effect. Using a priority would allow to always register a maintenance page under `/` with `prio 999` for example so that when no service is up the maintenance page is shown. I need to think about this a bit more.

**Comment by deuch** [comment]
Yep but playing with weight can be confusing sometimes :) A good named flag to define the route to go in case of no service is clearer to understand and setup :)

**Comment by tino** [comment]
`weight 0` doesn't work in my testing (v1.5.2), as it sets the weight to be balanced, as is also mentioned in the Route weight docs: https://github.com/fabiolb/fabio/wiki/Routing#route-weight
